app/components/windows/SourceProperties.vue.ts

```diff
@@ -35,8 +36,19 @@ export default class SourceProperties extends Vue {
   source = this.sourcesService.getSource(this.sourceId);
   properties: TFormData = [];
 
+  sourcesSubscription: Subscription;
+
   mounted() {
     this.properties = this.source ? this.source.getPropertiesFormData() : [];
+    this.sourcesSubscription = this.sourcesService.sourceRemoved.subscribe(source => {
```

Comment:
Does it work when you switch scene collection?

Reply:
Yup.  All of these events are fired even when sources are removed as part of swapping a scene collection.